core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java

```diff
@@ -151,6 +153,20 @@ public OAuth2Config setUserinfoUrl(String userinfoUrl)
         return this;
     }
 
+    @NotNull
+    public String getUserInfoRequestMethod()
+    {
+        return userInfoRequestMethod;
+    }
+
+    @Config("http-server.authentication.oauth2.userinfo-request-method")
+    @ConfigDescription("userinfo request method")
+    public OAuth2Config setUserInfoRequestMethod(String userInfoRequestMethod)
+    {
+        this.userInfoRequestMethod = userInfoRequestMethod;
```

Comment:
Please add
```java
@PostConstruct
public void validate()
{
}
```
ensuring that a valid HTTP method name was provided and a test for it.

Reply:
I added the method and verified the valid states using checkState